packages/server/lib/browsers/cdp_automation.ts

```diff
@@ -174,18 +176,36 @@ const ffToStandardResourceTypeMap: { [ff: string]: ResourceType } = {
   'webmanifest': 'manifest',
 }
 
+export interface CdpOptions {
+  sendDebuggerCommandFn: SendDebuggerCommand
+  onFn: OnFn
+  automation: Automation
+  experimentalSessionAndOrigin: boolean
+}
+
 export class CdpAutomation {
-  constructor (private sendDebuggerCommandFn: SendDebuggerCommand, onFn: OnFn, private automation: Automation) {
-    onFn('Network.requestWillBeSent', this.onNetworkRequestWillBeSent)
-    onFn('Network.responseReceived', this.onResponseReceived)
-    sendDebuggerCommandFn('Network.enable', {
+  sendDebuggerCommandFn: SendDebuggerCommand
+  automation: Automation
+  experimentalSessionAndOrigin: boolean
+
+  constructor (options: CdpOptions) {
+    this.sendDebuggerCommandFn = options.sendDebuggerCommandFn
+    this.automation = options.automation
+    this.experimentalSessionAndOrigin = options.experimentalSessionAndOrigin
+
+    options.onFn('Network.requestWillBeSent', this.onNetworkRequestWillBeSent)
+    options.onFn('Network.responseReceived', this.onResponseReceived)
+  }
+
+  async enable () {
+    await this.sendDebuggerCommandFn('Network.enable', {
       maxTotalBufferSize: 0,
       maxResourceBufferSize: 0,
       maxPostDataSize: 0,
     })
   }
 
-  private onNetworkRequestWillBeSent = (params: Protocol.Network.RequestWillBeSentEvent) => {
+  private onNetworkRequestWillBeSent = async (params: Protocol.Network.RequestWillBeSentEvent) => {
```

Comment:
this got `async` added but doesn't appear to need it?

```suggestion
  private onNetworkRequestWillBeSent = (params: Protocol.Network.RequestWillBeSentEvent) => {
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/793cac1d802316c88f926478174ba56cb8c8d031